frontend/app/(footer)/login/page.tsx

```diff
@@ -1,33 +1,40 @@
 "use client";
+import { Logger } from "@/observability/logger";
 import { handleEmailAndPasswordSignIn } from "@/services/src/auth/authService";
 import { Alert } from "@material-tailwind/react";
 import Link from "next/link";
 import { useRouter } from "next/navigation";
-import React, { useState } from "react";
+import React, { useState, useTransition } from "react";
 
 export default function Login() {
   const [userData, setUserData] = useState({
     email: "",
     password: "",
   });
   const [errorMessage, setErrorMessage] = useState("");
-  const [AlertStatus, setAlertStatus] = useState(false);
+  const [alertStatus, setAlertStatus] = useState(false);
+  const [isPending, startTransition] = useTransition();
   const router = useRouter();
+  const logger = new Logger("loginLogger");
 
   const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
+    handleAlertClose();
 
-    try {
-      const userCreated = await handleEmailAndPasswordSignIn(userData.email, userData.password);
-      if (userCreated) {
-        router.push("/dashboard?login=success"); // Redirect only if user creation is successful
-      } else {
-        console.error("User creation failed.");
+    startTransition(async () => {
+      try {
+        const userCreated = await handleEmailAndPasswordSignIn(userData.email, userData.password);
+        if (userCreated) {
+          router.push("/dashboard?login=success"); // Redirect only if user creation is successful
+        } else {
+          console.error("User creation failed.");
```

Comment:
Should we be alerting the user if their account creation has failed? This currently only logs the error to the console.

Reply:
Good catch, changed it to throw and catch the error instead